[CI] InferenceIngestIT.testPipelineIngest and testPathologicalPipelineCreationAndDeletion failures

<!--
Please fill out the following information, and ensure you have attempted
to reproduce locally
-->

**Build scan**:
https://gradle-enterprise.elastic.co/s/2kdpwkkrjc6r4

**Repro line**:
> ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTest' --tests "org.elasticsearch.xpack.ml.integration.InferenceIngestIT.testPipelineIngest" -Dtests.seed=F132D0E49ECDF4B3 -Dtests.security.manager=true -Dtests.locale=sr-ME -Dtests.timezone=Pacific/Midway -Druntime.java=8
>
> ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTest' --tests "org.elasticsearch.xpack.ml.integration.InferenceIngestIT.testPathologicalPipelineCreationAndDeletion" -Dtests.seed=F132D0E49ECDF4B3 -Dtests.security.manager=true -Dtests.locale=sr-ME -Dtests.timezone=Pacific/Midway -Druntime.java=8
>

**Reproduces locally?**:
No

**Applicable branches**:
* 7.x
* 7.9

**Failure history**:
<!--
Link to build stats and possible indication of when this started failing and how often it fails
<https://build-stats.elastic.co/app/kibana>
-->
These two tests seem only fail for v7 branches. It failed 8 times within last 60 days according to [build-stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-60d,mode:quick,to:now))&_a=(columns:!(build.branch),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b646ed00-7efc-11e8-bf69-63c8ef516157,key:build.project,negate:!f,params:(query:elasticsearch,type:phrase),type:phrase,value:elasticsearch),query:(match:(build.project:(query:elasticsearch,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b646ed00-7efc-11e8-bf69-63c8ef516157,key:build.branch,negate:!t,params:!(pull-request,pull-request-1,pull-request-2,pull-request-bwc),type:phrases,value:'pull-request,%20pull-request-1,%20pull-request-2,%20pull-request-bwc'),query:(bool:(minimum_should_match:1,should:!((match_phrase:(build.branch:pull-request)),(match_phrase:(build.branch:pull-request-1)),(match_phrase:(build.branch:pull-request-2)),(match_phrase:(build.branch:pull-request-bwc))))))),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'InferenceIngestIT%20cache_miss_count'),sort:!(process.time-start,desc))).

When they fail, there are always failures of `ClassificationIT  in the same build scan with "ClusterHealthResponse has timed out" error. Maybe they are related, like one is the cause of the other.

I also noticed we have a previous issue (#54786) for testPipelineIngest but the failure message is different. So I am opening a new issue.

**Failure excerpt**:
```

java.lang.AssertionError: |  
-- | --
  | Expected: a string containing "\"cache_miss_count\":3" |  
  | but: was "{"count":1,"trained_model_stats":[{"model_id":"test_classification","pipeline_count":0,"inference_stats":{"failure_count":0,"inference_count":10,"cache_miss_count":2,"missing_all_fields_count":0,"timestamp":1598400435615}}]}" |  

at __randomizedtesting.SeedInfo.seed([F132D0E49ECDF4B3:536E46D26C7D13F0]:0) |  
-- | --
  |   | •••
  |   | at org.elasticsearch.xpack.ml.integration.InferenceIngestIT.lambda$testPipelineIngest$1(InferenceIngestIT.java:178) |  
  |   | •••
  |   | at org.elasticsearch.xpack.ml.integration.InferenceIngestIT.testPipelineIngest(InferenceIngestIT.java:172) |  
  |   | •••


```

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
@@ -192,7 +192,7 @@ public class InferenceIngestIT extends ESRestTestCase {
         assertBusy(() -> {
             try {
                 assertStatsWithCacheMisses(classificationModelId, 10L);
-                assertStatsWithCacheMisses(regressionModelId, 10L);
+                assertStatsWithCacheMisses(regressionModelId, 15L);
             } catch (ResponseException ex) {
                 //this could just mean shard failures.
                 fail(ex.getMessage());

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
@@ -142,25 +142,8 @@ public class InferenceIngestIT extends ESRestTestCase {
         assertThat(EntityUtils.toString(searchResponse.getEntity()), containsString("\"value\":10"));
         assertBusy(() -> {
             try {
-                Response statsResponse = client().performRequest(new Request("GET",
-                    "_ml/trained_models/" + classificationModelId + "/_stats"));
-                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
-                    assertThat(response.getTrainedModelStats(), hasSize(1));
-                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
-                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
-                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
-                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
-                }
-                statsResponse = client().performRequest(new Request("GET", "_ml/trained_models/" + regressionModelId + "/_stats"));
-                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
-                    assertThat(response.getTrainedModelStats(), hasSize(1));
-                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
-                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
-                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
-                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
-                }
+                assertStatsWithCacheMisses(classificationModelId, 10L);
+                assertStatsWithCacheMisses(regressionModelId, 10L);
             } catch (ResponseException ex) {
                 //this could just mean shard failures.
                 fail(ex.getMessage());
@@ -208,30 +191,26 @@ public class InferenceIngestIT extends ESRestTestCase {
 
         assertBusy(() -> {
             try {
-                Response statsResponse = client().performRequest(new Request("GET",
-                    "_ml/trained_models/" + classificationModelId + "/_stats"));
-                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
-                    assertThat(response.getTrainedModelStats(), hasSize(1));
-                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
-                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
-                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
-                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
-                }
-                statsResponse = client().performRequest(new Request("GET", "_ml/trained_models/" + regressionModelId + "/_stats"));
-                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
-                    assertThat(response.getTrainedModelStats(), hasSize(1));
-                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
-                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
-                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(15L));
-                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
-                }
+                assertStatsWithCacheMisses(classificationModelId, 10L);
+                assertStatsWithCacheMisses(regressionModelId, 10L);
             } catch (ResponseException ex) {
                 //this could just mean shard failures.
                 fail(ex.getMessage());
             }
         }, 30, TimeUnit.SECONDS);
+    }
+
+    public void assertStatsWithCacheMisses(String modelId, long inferenceCount) throws IOException {
+        Response statsResponse = client().performRequest(new Request("GET",
+            "_ml/trained_models/" + modelId + "/_stats"));
+        try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
+            GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
+            assertThat(response.getTrainedModelStats(), hasSize(1));
+            TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
+            assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
+            assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(inferenceCount));
+            assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
+        }
     }
 
     public void testSimulate() throws IOException {

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
@@ -10,6 +10,8 @@ import org.elasticsearch.client.Request;
 import org.elasticsearch.client.RequestOptions;
 import org.elasticsearch.client.Response;
 import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.ml.GetTrainedModelsStatsResponse;
+import org.elasticsearch.client.ml.inference.TrainedModelStats;
 import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
@@ -17,7 +19,9 @@ import org.elasticsearch.common.xcontent.NamedXContentRegistry;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
 import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.elasticsearch.test.ExternalTestCluster;
@@ -38,6 +42,8 @@ import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
 import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
 
 /**
  * This is a {@link ESRestTestCase} because the cleanup code in {@link ExternalTestCluster#ensureEstimatedStats()} causes problems
@@ -135,13 +141,17 @@ public class InferenceIngestIT extends ESRestTestCase {
             try {
                 Response statsResponse = client().performRequest(new Request("GET",
                     "_ml/trained_models/" + classificationModelId + "/_stats"));
-                String response = EntityUtils.toString(statsResponse.getEntity());
-                assertThat(response, containsString("\"inference_count\":10"));
-                assertThat(response, containsString("\"cache_miss_count\":30"));
+                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
+                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
+                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
+                }
                 statsResponse = client().performRequest(new Request("GET", "_ml/trained_models/" + regressionModelId + "/_stats"));
-                response = EntityUtils.toString(statsResponse.getEntity());
-                assertThat(response, containsString("\"inference_count\":10"));
-                assertThat(response, containsString("\"cache_miss_count\":30"));
+                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
+                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
+                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
+                }
             } catch (ResponseException ex) {
                 //this could just mean shard failures.
                 fail(ex.getMessage());
@@ -191,18 +201,17 @@ public class InferenceIngestIT extends ESRestTestCase {
             try {
                 Response statsResponse = client().performRequest(new Request("GET",
                     "_ml/trained_models/" + classificationModelId + "/_stats"));
-                String response = EntityUtils.toString(statsResponse.getEntity());
-                assertThat(response, containsString("\"inference_count\":10"));
-                assertThat(response, containsString("\"cache_miss_count\":3"));
+                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
+                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
+                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
+                }
                 statsResponse = client().performRequest(new Request("GET", "_ml/trained_models/" + regressionModelId + "/_stats"));
-                response = EntityUtils.toString(statsResponse.getEntity());
-                assertThat(response, containsString("\"inference_count\":15"));
-                assertThat(response, containsString("\"cache_miss_count\":3"));
-                // can get both
-                statsResponse = client().performRequest(new Request("GET", "_ml/trained_models/_stats"));
-                String entityString = EntityUtils.toString(statsResponse.getEntity());
-                assertThat(entityString, containsString("\"inference_count\":15"));
-                assertThat(entityString, containsString("\"inference_count\":10"));
+                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
+                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(15L));
+                    assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
+                }
             } catch (ResponseException ex) {
                 //this could just mean shard failures.
                 fail(ex.getMessage());

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java
@@ -42,8 +42,11 @@ import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
 import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
 
 /**
  * This is a {@link ESRestTestCase} because the cleanup code in {@link ExternalTestCluster#ensureEstimatedStats()} causes problems
@@ -142,13 +145,19 @@ public class InferenceIngestIT extends ESRestTestCase {
                 Response statsResponse = client().performRequest(new Request("GET",
                     "_ml/trained_models/" + classificationModelId + "/_stats"));
                 try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
+                    assertThat(response.getTrainedModelStats(), hasSize(1));
+                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
                     assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
                     assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
                 }
                 statsResponse = client().performRequest(new Request("GET", "_ml/trained_models/" + regressionModelId + "/_stats"));
                 try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
+                    assertThat(response.getTrainedModelStats(), hasSize(1));
+                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
                     assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
                     assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
                 }
@@ -202,13 +211,19 @@ public class InferenceIngestIT extends ESRestTestCase {
                 Response statsResponse = client().performRequest(new Request("GET",
                     "_ml/trained_models/" + classificationModelId + "/_stats"));
                 try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
+                    assertThat(response.getTrainedModelStats(), hasSize(1));
+                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
                     assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(10L));
                     assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
                 }
                 statsResponse = client().performRequest(new Request("GET", "_ml/trained_models/" + regressionModelId + "/_stats"));
                 try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
-                    TrainedModelStats trainedModelStats = GetTrainedModelsStatsResponse.fromXContent(parser).getTrainedModelStats().get(0);
+                    GetTrainedModelsStatsResponse response = GetTrainedModelsStatsResponse.fromXContent(parser);
+                    assertThat(response.getTrainedModelStats(), hasSize(1));
+                    TrainedModelStats trainedModelStats = response.getTrainedModelStats().get(0);
+                    assertThat(trainedModelStats.getInferenceStats(), is(notNullValue()));
                     assertThat(trainedModelStats.getInferenceStats().getInferenceCount(), equalTo(15L));
                     assertThat(trainedModelStats.getInferenceStats().getCacheMissCount(), greaterThan(0L));
                 }

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/LocalModel.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/LocalModel.java
@@ -58,14 +58,14 @@ public class LocalModel implements Closeable {
     private final AtomicLong referenceCount;
 
     LocalModel(String modelId,
-                      String nodeId,
-                      InferenceDefinition trainedModelDefinition,
-                      TrainedModelInput input,
-                      Map<String, String> defaultFieldMap,
-                      InferenceConfig modelInferenceConfig,
-                      License.OperationMode licenseLevel,
-                      TrainedModelStatsService trainedModelStatsService,
-                      CircuitBreaker trainedModelCircuitBreaker) {
+               String nodeId,
+               InferenceDefinition trainedModelDefinition,
+               TrainedModelInput input,
+               Map<String, String> defaultFieldMap,
+               InferenceConfig modelInferenceConfig,
+               License.OperationMode licenseLevel,
+               TrainedModelStatsService trainedModelStatsService,
+               CircuitBreaker trainedModelCircuitBreaker) {
         this.trainedModelDefinition = trainedModelDefinition;
         this.modelId = modelId;
         this.fieldNames = new HashSet<>(input.getFieldNames());

